tests/unit/processor/domain_resolver/test_domain_resolver.py

```diff
@@ -28,21 +27,159 @@ class TestDomainResolver(BaseProcessorTestCase):
         "tree_config": "tests/testdata/unit/shared_data/tree_config.json",
     }
 
-    def test_domain_to_ip_resolved_and_added(self, monkeypatch):
-        def mockreturn(domain):
-            if domain == "google.de":
-                return "1.2.3.4"
-            else:
-                return None
+    @mock.patch("socket.gethostbyname", return_value="1.2.3.4")
+    def test_domain_to_ip_resolved_and_added(self, _):
+        rule = {
+            "filter": "fqdn",
+            "domain_resolver": {"source_url_or_domain": "fqdn"},
+            "description": "",
+        }
+        self._load_specific_rule(rule)
+        document = {"fqdn": "google.de"}
+        expected = {"fqdn": "google.de", "resolved_ip": "1.2.3.4"}
+        self.object.process(document)
+        assert document == expected
+
+    @mock.patch("socket.gethostbyname", return_value="1.2.3.4")
+    def test_domain_to_ip_resolved_and_added_from_cache(self, _):
```

Comment:
How does this test know that it should resolve the domain from the cache? The test is actually exactly the same as the previous one `test_domain_to_ip_resolved_and_added`. Does this test maybe miss a second `self.object.process(document)` call?

Reply:
rewrote test